landlab/ca/cfuncs.pyx

```diff
@@ -522,21 +515,23 @@ cpdef push_transitions_to_event_queue(
 @cython.boundscheck(True)
 @cython.wraparound(False)
 cdef void update_link_state(
-    DTYPE_INT_t link, DTYPE_INT_t new_link_state,
-    DTYPE_t current_time,
-    np.ndarray[DTYPE_INT8_t, ndim=1] bnd_lnk,
-    np.ndarray[DTYPE_INT_t, ndim=1] node_state,
-    np.ndarray[DTYPE_INT_t, ndim=1] node_at_link_tail,
-    np.ndarray[DTYPE_INT_t, ndim=1] node_at_link_head,
-    np.ndarray[DTYPE_INT8_t, ndim=1] link_orientation,
-    DTYPE_INT_t num_node_states,
-    DTYPE_INT_t num_node_states_sq,
-    np.ndarray[DTYPE_INT_t, ndim=1] link_state,
-    np.ndarray[DTYPE_INT_t, ndim=1] n_xn, event_queue,
-    np.ndarray[DTYPE_t, ndim=1] next_update,
-    np.ndarray[DTYPE_INT_t, ndim=2] xn_to,
-    np.ndarray[DTYPE_t, ndim=2] xn_rate,
-    np.ndarray[DTYPE_INT8_t, ndim=2] xn_propswap,
+    long link,
+    long new_link_state,
+    double current_time,
+    int8_t [:] bnd_lnk,
+    const id_t [:] node_state,
+    const id_t [:] node_at_link_tail,
+    const id_t [:] node_at_link_head,
+    const int8_t [:] link_orientation,
+    long num_node_states,
+    long num_node_states_sq,
+    id_t [:] link_state,
+    id_t [:] n_xn,
+    event_queue,
```

Comment:
With this and the following untyped parameters you flagged, I should probably try to add types for them. Since they were untyped before, however, it shouldn't slow things down.